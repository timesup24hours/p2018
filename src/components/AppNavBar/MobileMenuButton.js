import React from 'react';
import styles from './index.module.css';
import { MaterialDesignBurgerMenu } from '../MaterialDesignBurgerMenu';

export default React.forwardRef((props, ref) => (
  <div className={styles.menuButton} ref={ref}>
    <MaterialDesignBurgerMenu
      customhandleOnClick={props.customhandleOnClick}
      open={props.open}
    />
  </div>
));
