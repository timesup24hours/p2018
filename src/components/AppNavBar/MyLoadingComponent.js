import React from 'react';
import Loading from '../Loading';

export default ({ isLoading, error }) => {
  // Handle the loading state
  if (isLoading) {
    return (
      <div className="center">
        <Loading />
      </div>
    );
  }
  // Handle the error state
  else if (error) {
    return (
      <div className="center">Sorry, there was a problem loading the page.</div>
    );
  } else {
    return null;
  }
};
