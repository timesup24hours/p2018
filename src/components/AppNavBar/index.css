.nav_ul {
  display: grid;
  grid-template-columns: 1fr 1fr 1fr 1fr;
  grid-template-areas: 'nav_ul_li_1 nav_ul_li_2 nav_ul_li_3 nav_ul_li_4';
  height: var(--nav-height);
  grid-gap: 20px;
  /* border: 1px solid black; */
  width: 80%;
  margin: 0 auto;
}

.nav_ul li {
  /* border: 1px solid red; */
  list-style: none;
  height: 100%;
  display: block;
}

.nav_ul li a {
  text-decoration-line: none;
  display: block;
  height: 100%;
  width: 100%;
  text-align: center;
  line-height: calc(var(--nav-height));
}

.nav_ul_li_1 {
}
.nav_ul_li_2 {
}
.nav_ul_li_3 {
}
.nav_ul_li_4 {
}

@media screen and (max-width: 660px) {
  .nav_ul {
    display: grid;
    grid-template-columns: 100%;
    grid-template-areas:
      'nav_ul_li_1'
      'nav_ul_li_2'
      'nav_ul_li_3'
      'nav_ul_li_4';
    height: var(--nav-height-660px);
    grid-gap: 2px;
    /* border: 1px solid red; */
    width: 100%;
    margin: 0 auto;
  }
}
