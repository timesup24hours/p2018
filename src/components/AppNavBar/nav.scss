:root {
  --nav-line-top: 0;
}
.Nav_nav {
  /* border: 1px solid black; */
  z-index: 9000;
  position: fixed;
  width: 100%;
  background: rgba(255, 255, 255, 0.9);
  opacity: 0.9;
  /* box-shadow: 0 5px 15px 2px rgba(0, 0, 0, 0.5); */
  filter: alpha(opacity=30);

  .nav_ul {
    display: grid;
    grid-template-columns: 1fr 1fr 1fr 1fr;
    grid-template-areas: 'nav_ul_li_1 nav_ul_li_2 nav_ul_li_3 nav_ul_li_4';
    height: var(--nav-height);
    /* grid-gap: 20px; */
    /* border: 1px solid black; */
    width: 100%;
    margin: 0 auto;
    padding-left: 60%;
    padding-right: 2%;
    position: relative;
    /* font-weight: bold; */

    /* position: fixed; */
    /* z-index: 1000; */
  }

  .nav_ul li {
    list-style: none;
    height: 100%;
    display: flex;
    justify-content: center;
    align-items: center;
  }

  .a {
    color: black;
    margin: 0 auto;
    text-decoration-line: none;
    display: block;
    height: 100%;
    width: 100%;
    text-align: center;

    line-height: calc(var(--nav-height));

    /* transition: 0.5s; */
  }
  .a.active span {
    border-bottom: 2px solid rgba(117, 72, 159, 0.5);
  }

  .nav_ul li a:visited {
    color: none;
    text-decoration: none;
  }
  .active {
    /* color: blueviolet; */
    color: #75489f;
    /* border-bottom: 2px solid rgba(117, 72, 159, 0.5); */
    /* text-decoration: underline;
  text-decoration-color: #75489f; */
  }

  .navActive {
    display: block;
  }
  .navInactive {
    display: block;
  }

  .menuButton {
    display: none;
    z-index: 10000;
    border-radius: 50%;
    /* background: rgba(255, 255, 255, 1); */
    background: rgba(117, 72, 159, 0.8);
  }
  .menuButton:hover {
    cursor: pointer;
  }

  .nav_ul_li_1 {
  }
  .nav_ul_li_2 {
  }
  .nav_ul_li_3 {
  }
  .nav_ul_li_4 {
  }
  /* .pageWrapper 
  position: absolute;
  top: var(--nav-height);
} */

  @media screen and (max-width: 660px) {
    .nav {
      display: none;
      opacity: 80%;
    }
    .nav_ul {
      display: grid;
      grid-template-columns: 100%;
      grid-template-areas:
        'nav_ul_li_1'
        'nav_ul_li_2'
        'nav_ul_li_3'
        'nav_ul_li_4';
      height: var(--nav-height-660px);
      grid-gap: 2px;
      width: 100%;
      margin: 0 auto;
      padding-left: 0;
      padding-right: 0;
    }
    /* .nav_ul {
    display: grid;
    grid-template-columns: 100%;
    grid-template-areas:
      'nav_ul_li_1'
      'nav_ul_li_2'
      'nav_ul_li_3'
      'nav_ul_li_4';
    height: var(--nav-height-660px);
    grid-gap: 2px;
    width: 100%;
    margin: 0;
    padding-left: 0;
    padding-right: 0;
    position: absolute;
    top: 0;
    left: 90%;
    box-shadow: 0 5px 15px 2px rgba(0, 0, 0, 0.5);
  } */
    .menuButton {
      display: block;
      z-index: 10000;
      height: 50px;
      width: 50px;
      position: fixed;
      top: 8px;
      left: 8px;
      /* border-radius: 4px; */
    }
    .menuButton div {
      transform: scale(0.8);
    }
    .navActive {
      display: block;
    }
  }
  .nav_line {
    display: inline-block;
    transition: all 0.3s ease-in-out;
    position: absolute;
    background-color: #75489f;
    width: var(--nav-line-width);
    height: 4px;
    top: var(--nav-line-bottom);
    left: var(--nav-line-left);
  }
}

.navActive {
  display: block;
}
.navInactive {
  display: none;
}
