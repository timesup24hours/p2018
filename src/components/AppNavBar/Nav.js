import React, { useEffect, useState, useRef } from 'react';
import { NavLink as Link } from 'react-router-dom';

import styles from './index.module.css';
import './nav.scss';

export default React.forwardRef((props, ref) => {
  const lineRef = useRef(null);
  const [style, setStyle] = useState({ left: 0, width: 0 });
  const getRect = (ele: Element | null) => {
    if (!ele) return 0;
    const rect = ele.getBoundingClientRect();
    return rect;
  };
  useEffect(() => {
    const activeNode: Element | null = document.querySelector(
      `.${styles.active}`
    );
    const left = getRect(activeNode).left;
    document.documentElement.style.setProperty('--nav-line-top', `${left}px`);
  });
  const handleNavOnClick = (event: React.MouseEvent) => {
    const ele: Element = event.target;
    if (!ele.classList.contains(styles.a)) return;
    const rect = getRect(ele.children[0]);
    setStyle(rect);
    // console.log('line: ', (lineRef.current.style.left = left));
    // document.documentElement.style.setProperty('--nav-line-top', `${left}px`);
  };
  return (
    <nav className={styles.nav} ref={ref}>
      <ul className={styles.nav_ul}>
        <li className={styles.nav_ul_li_1}>
          <Link
            exact
            className={styles.a}
            activeClassName={styles.active}
            onClick={event => {
              props.CustomhandleOnClick(event);
              handleNavOnClick(event);
            }}
            to="/"
          >
            <span>HOME</span>
          </Link>
        </li>
        <li className={styles.nav_ul_li_2}>
          <Link
            className={styles.a}
            activeClassName={styles.active}
            to="/my_notes"
            onClick={event => {
              props.CustomhandleOnClick(event);
              handleNavOnClick(event);
            }}
          >
            <span>NOTES</span>
          </Link>
        </li>
        <li className={styles.nav_ul_li_3}>
          <Link
            className={styles.a}
            activeClassName={styles.active}
            to="/css_effects"
            onClick={event => {
              props.CustomhandleOnClick(event);
              handleNavOnClick(event);
            }}
          >
            <span>UI</span>
          </Link>
        </li>
        <li className={styles.nav_ul_li_4}>
          <Link
            className={styles.a}
            activeClassName={styles.active}
            to="/contact_info"
            onClick={event => {
              props.CustomhandleOnClick(event);
              handleNavOnClick(event);
            }}
          >
            <span>CONTACT</span>
          </Link>
        </li>
        <div
          ref={lineRef}
          className={styles.line}
          style={{ left: style.left, width: style.width }}
        />
      </ul>
    </nav>
  );
});
