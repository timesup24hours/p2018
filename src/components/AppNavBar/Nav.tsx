import React, { useEffect, useState, useRef, MouseEvent } from 'react';
import { NavLink as Link } from 'react-router-dom';

// import styles from './index.module.css';
import './nav.scss';

interface rectObject {
  left: number;
  top: number;
  width: number;
  bottom: number;
}

interface NavProps {
  customhandleOnClick: (event: MouseEvent) => void;
}

export default React.forwardRef(
  (props: NavProps, ref?: React.Ref<HTMLDivElement>) => {
    const lineRef = useRef<HTMLDivElement>(null);
    const [style, setStyle] = useState({
      left: 0,
      width: 0,
      bottom: 36,
      top: 0
    });
    const getRect = (ele: Element | null): ClientRect | rectObject => {
      if (!ele) return { left: 0, width: 0, bottom: 36, top: 0 };
      const rect = ele.getBoundingClientRect();
      return rect;
    };
    useEffect(() => {
      const activeNode: Element | null = document.querySelector(`.active`);
      if (!activeNode) return;
      const rect = getRect(activeNode!.children[0]);
      document.documentElement.style.setProperty(
        '--nav-line-left',
        `${rect.left}px`
      );
      document.documentElement.style.setProperty(
        '--nav-line-width',
        `${rect.width}px`
      );
      document.documentElement.style.setProperty(
        '--nav-line-bottom',
        `${rect.bottom - 1}px`
      );
    });
    // const handleNavOnClick = (event: React.MouseEvent) => {
    //   const ele = event.currentTarget!;
    //   if (!ele.classList.contains('a')) return;
    //   const rect = getRect(ele.children[0]);
    //   setStyle(rect);
    //   document.documentElement.style.setProperty(
    //     '--nav-line-top',
    //     `${rect.left}px`
    //   );
    // };
    return (
      <nav className="Nav_nav" ref={ref}>
        <ul className="nav_ul">
          <li className="nav_ul_li_1">
            <Link
              exact
              className="a"
              activeClassName="active"
              onClick={event => {
                props.customhandleOnClick(event);
                // handleNavOnClick(event);
              }}
              to="/"
            >
              <span>HOME</span>
            </Link>
          </li>
          <li className="nav_ul_li_2">
            <Link
              className="a"
              activeClassName="active"
              to="/my_notes"
              onClick={event => {
                props.customhandleOnClick(event);
                // handleNavOnClick(event);
              }}
            >
              <span>NOTES</span>
            </Link>
          </li>
          <li className="nav_ul_li_3">
            <Link
              className="a"
              activeClassName="active"
              to="/css_effects"
              onClick={event => {
                props.customhandleOnClick(event);
                // handleNavOnClick(event);
              }}
            >
              <span>UI</span>
            </Link>
          </li>
          <li className="nav_ul_li_4">
            <Link
              className="a"
              activeClassName="active"
              to="/contact_info"
              onClick={event => {
                props.customhandleOnClick(event);
                // handleNavOnClick(event);
              }}
            >
              <span>CONTACT</span>
            </Link>
          </li>
          <div
            ref={lineRef}
            className="nav_line"
            // style={{ left: style.left, width: style.width }}
          />
        </ul>
      </nav>
    );
  }
);
