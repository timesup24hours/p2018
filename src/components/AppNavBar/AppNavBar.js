import React, { Component } from 'react';
import {
  // BrowserRouter as Router // for production
  HashRouter as Router // for github free host hard reload
} from 'react-router-dom';
// import styled from 'styled-components';

import styles from './index.css';

import MobileMenuButton from './MobileMenuButton';
import Nav from './Nav';
import Routes from './Routes';

// const MenuButtonContainer = styled.div`
//   display: none;

//   &:hover {
//     cursor: pointer;
//   }

//   @media (max-width: 600px) {
//     display: block;
//     z-index: 10000;
//     height: 50px;
//     width: 50px;
//     position: fixed;
//     top: 6px;
//     left: 6px;
//     border-radius: 4px;
//   }
// `;

class AppNavBar extends Component {
  state = {
    isOpen: false
    // isMobile: false
  };
  refNav = React.createRef();
  MobileMenuButton = React.createRef();

  // componentDidMount() {
  //   // console.log(window.innerHeight < 700);
  //   let testExp = new RegExp(
  //     'Android|webOS|iPhone|iPad|' +
  //       'BlackBerry|Windows Phone|' +
  //       'Opera Mini|IEMobile|Mobile',
  //     'i'
  //   );
  //   this.setState({ isMobile: testExp.test(navigator.userAgent) });
  // }

  toggle = () => {
    this.setState({
      isOpen: !this.state.isOpen
    });
  };

  CustomhandleOnClick = e => {
    if (getComputedStyle(this.MobileMenuButton.current).display === 'block') {
      this.refNav.current.classList.toggle(styles.navActive);
      this.toggle();
    }
  };

  render() {
    const { isOpen } = this.state;
    return (
      <Router>
        <div>
          <MobileMenuButton
            open={isOpen}
            CustomhandleOnClick={this.CustomhandleOnClick}
            ref={this.MobileMenuButton}
          />
          <Nav
            CustomhandleOnClick={this.CustomhandleOnClick}
            ref={this.refNav}
          />
          <Routes />
        </div>
      </Router>
    );
  }
}

export default AppNavBar;
