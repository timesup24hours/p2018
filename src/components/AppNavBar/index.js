// import React from 'react';
import AppNavBar from './AppNavBar';

export default AppNavBar;
