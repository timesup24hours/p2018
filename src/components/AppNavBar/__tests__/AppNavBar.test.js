import React from 'react';
import { shallow, mount } from 'enzyme';
import AppNavBar from '../AppNavBar';

describe('AppNavBar', () => {
  it('should render correctly in "debug" mode', () => {
    const component = shallow(<AppNavBar debug />);

    expect(component).toMatchSnapshot();
  });

  it('should be contain active className when route clicked', () => {
    const wrapper = mount(<AppNavBar />);
    expect(wrapper.state().isOpen).toBeTruthy();

    const instance = wrapper.instance();
    instance.toggle();
    expect(wrapper.state().isOpen).toBeFalsy();

    // console.log(component.find('[data-testid="my_notes"]').debug());
    // const link = component.find('[data-testid="my_notes"]').at(0);

    // link.simulate('click');

    // expect(component.find('.active').length).toBe(1);
    // console.log(component.find('.active').debug());
    // // console.log(link.debug());
    // // expect(component).toMatchSnapshot();
    // // component.unmount();
    // expect(link.hasClass('a')).toBe(true);
  });

  it('toggle method should change the state', () => {
    const wrapper = mount(<AppNavBar />);

    const instance = wrapper.instance();
    instance.toggle();
    expect(wrapper.state().isOpen).toBeFalsy();
  });
});
