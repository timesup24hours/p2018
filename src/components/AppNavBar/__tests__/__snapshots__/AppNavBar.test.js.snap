// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppNavBar should render correctly in "debug" mode 1`] = `
<HashRouter>
  <div>
    <ForwardRef
      customhandleOnClick={[Function]}
      open={true}
    />
    <ForwardRef
      customhandleOnClick={[Function]}
    />
    <_default />
  </div>
</HashRouter>
`;
